lib/bolt/transport/sudoable/connection_helper.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+require 'bolt/transport/sudoable/file_commands'
+
+module Bolt
+  module Transport
+    class Sudoable < Base
+      class ConnectionHelper
+        def initialize(target)
+          @target = target
+          @run_as = nil
+          @logger = Logging.logger[@target.host]
+        end
+
+        def make_executable(path)
+          result = execute(['chmod', 'u+x', path])
+          if result.exit_code != 0
+            message = "Could not make file '#{path}' executable: #{result.stderr.string}"
+            raise Bolt::Node::FileError.new(message, 'CHMOD_ERROR')
+          end
+        end
+
+        def make_tempdir
+          tmpdir = @target.options.fetch('tmpdir', '/tmp')
+          tmppath = "#{tmpdir}/#{SecureRandom.uuid}"
+          command = ['mkdir', '-m', 700, tmppath]
+
+          result = execute(command)
+          if result.exit_code != 0
+            raise Bolt::Node::FileError.new("Could not make tempdir: #{result.stderr.string}", 'TEMPDIR_ERROR')
+          end
+          path = tmppath || result.stdout.string.chomp
+          Sudoable::FileCommands.new(self, path)
+        end
+
+        # A helper to create and delete a tempdir on the remote system. Yields the
+        # directory name.
+        def with_tempdir
```

Comment:
This delete does not get run with the elevated privileges. I have not had a chance to dig in, but it also seems random if the error is even surfaced for `script run`. Sometime I get 
```
cas@cas-ThinkPad-T460p:~/working_dir/bolt/Boltdir$ bolt script run test.sh -n localhost --run-as root --sudo-password $SECRET
Started on localhost...
Failed to clean up tempdir '/tmp/57a91557-a963-4787-92b1-bd8e13d451cf': rm: cannot remove '/tmp/57a91557-a963-4787-92b1-bd8e13d451cf': Operation not permitted

Finished on localhost:
  STDOUT:
    hi
Successful on 1 node: localhost
Ran on 1 node in 0.06 seconds
```
and others the message does not surface. For `run task` it always fails
```
cas@cas-ThinkPad-T460p:~/working_dir/bolt/Boltdir$ bolt task run sample::noop -n localhost --run-as root --sudo-password $SECRET
Started on localhost...
Failed to clean up tempdir '/tmp/f294dbd1-3df6-453b-a772-f620b59faad1': rm: cannot remove '/tmp/f294dbd1-3df6-453b-a772-f620b59faad1': Operation not permitted

Failed on localhost:
  Broken pipe
Failed on 1 node: localhost
Ran on 1 node in 0.11 seconds
```